### PR TITLE
MODGOBI-73- Product Type ID to UUID

### PR DESCRIPTION
**Pull request by KVupp**
https://issues.folio.org/browse/MODGOBI-73
## Purpose
Change Product type Id to UUID, to  handle multiple types in inventory while creating an instance

## Approach
1. Add a lookup method to identifier-types
2. Set the UUID while creating the ORDER
3. Use default type ISBN to fetch UUID, and fall back to first one if not present(which is highly unlikely)
4. added test cases to exercise the scenario

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java [hunk 1]
```diff
@@ -1211,7 +1259,31 @@ private void handlePutOrderById(RoutingContext ctx) {
...
+      logger.info("got productTypes request: {}", ctx.request()
+        .query());
+      String getByIdInstruction = ctx.request()
+        .getHeader(MOCK_OKAPI_GET_IDENTIFIER_HEADER);
+
+      JsonObject productTypes = new JsonObject();
+      addServerRqRsData(HttpMethod.GET, IDENTIFIER_TYPES, productTypes);
 
+      if (MOCK_INSTRUCTION_FAIL_PRODUCTYPE.equals(getByIdInstruction) && ctx.request().query().contains("ISBN")) {
+        productTypes.put("identifierTypes",new JsonArray()).put(TOTAL_RECORDS, 0);
```
Comment (craigmcnally) [inline src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java]: space after comma

**Comment by craigmcnally** [comment]
This is somewhat confusing but it isn't really productTypeId, but rather productIdTypeId; that is, it's the id of the record representing the type of productId.

If productIdTypeId is too confusing, maybe just productIdType or identifierTypeId

**Comment by craigmcnally** [comment]
That said, from a technical perspective this LGTM.  I'm going to approve and let you decide how to best handle the naming... Addressing this is relatively minor and can be deferred to later if you choose.

File: src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java [hunk 2]
```diff
@@ -282,6 +283,29 @@ public PostGobiOrdersHelper(HttpClientInterface httpClient,
       });
   }
 
+  /**
+   * Use the provided product type.If the specified type can't be found, fallback using the first one listed
+   *
+   * @param productType
+   * @return
```
Comment (javalkar-varun) [inline src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java]: returning what in javadocs?

File: src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java [hunk 3]
```diff
@@ -223,7 +226,8 @@ public final void testPostGobiXMLContent(TestContext context) throws Exception {
         .contentType("application/xml");
 
     Map<String, List<JsonObject>> column = MockServer.serverRqRs.column(HttpMethod.GET);
-    assertEquals(5, column.size());
+    //  // Listed Electronic Monograph has to get the Product type ID so there will be an additional call made
```
Comment (javalkar-varun) [inline src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java]: nitpick: Redundant `//`

**Comment by KVupp** [comment]
> This is somewhat confusing but it isn't really productTypeId, but rather productIdTypeId; that is, it's the id of the record representing the type of productId.
> 
> If productIdTypeId is too confusing, maybe just productIdType or identifierTypeId

yea I was thinking about that too, but like you said, productIdTypeId is a little twisted, I'll go with productIdType to keep it simple

**Pull request merged**